samples/nrf9160/serial_lte_modem/src/tcpip_proxy/slm_at_tcp_proxy.c

```diff
@@ -414,13 +414,9 @@ static void tcp_thread_func(void *p1, void *p2, void *p3)
 				continue;
 			}
 			if (slm_util_hex_check(data, ret)) {
-				char *data_hex = k_malloc(ret * 2);
+				char data_hex[ret * 2];
```

Comment:
I'm just a bit concerned about the stack usage (here and in other places). In worst case scenario we'll allocate 3 * NET_IPV4_MTU (1728 bytes) just for the `data` and `data_hex`. Probably we should consider switching to static buffers instead to avoid potential overflows? Or keep the stack size `NET_IPV4_MTU` dependend (but increase from `+ NET_IPV4_MTU` to `+ 3 * NET_IPV4_MTU`). Personally I would prefer the former since it's a simpler, error-proof solution.

Reply:
Got it, will use static buffer for TCP/UDP receive thread
In case of send (system work context), will keep using stack